p2p/transport_test.go

```diff
@@ -534,13 +514,13 @@ func TestTransportHandshake(t *testing.T) {
 		}
 
 		go func(c net.Conn) {
-			_, err := cdc.MarshalBinaryWriter(c, peerNodeInfo)
+			_, err := cdc.MarshalBinaryWriter(c, peerNodeInfo.(DefaultNodeInfo))
```

Comment:
If we registered the NodeInfo interface with amino, we wouldn't have to do this, and we'd be able to adopt new NodeInfo types. But we probably don't want amino baked in there, so we'll need a better strategy for getting the concrete NodeInfo type than a type assertion in the depths of the code like this.